src/components/queryBuilder/views/LogsQueryBuilder.tsx

```diff
@@ -0,0 +1,138 @@
+import { ColumnsEditor } from '../ColumnsEditor';
+import { BuilderMode, Filter, TableColumn, OrderBy, QueryBuilderOptions, SelectedColumn, ColumnHint } from 'types/queryBuilder';
+import { ColumnSelect } from '../ColumnSelect';
+import { Switch } from '../Switch';
+import { OtelVersionSelect } from '../OtelVersionSelect';
+import { OrderByEditor } from '../OrderByEditor';
+import { LimitEditor } from '../LimitEditor';
+import { FiltersEditor } from '../FilterEditor';
+import allLabels from 'labels';
+import { getColumnByHint } from 'components/queryBuilder/utils';
+
+interface LogsQueryBuilderProps {
+  allColumns: TableColumn[];
+  builderOptions: QueryBuilderOptions,
+  onBuilderOptionsChange: (builderOptions: QueryBuilderOptions) => void;
+}
+
+export const LogsQueryBuilder = (props: LogsQueryBuilderProps) => {
+  const { allColumns, builderOptions, onBuilderOptionsChange } = props;
+  const [otelEnabled, setOtelEnabled] = useState<boolean>(false);
+  const [otelVersion, setOtelVersion] = useState<string>('');
+  const [selectedColumns, setSelectedColumns] = useState<SelectedColumn[]>([]);
+  const [timeColumn, setTimeColumn] = useState<SelectedColumn>();
+  const [logLevelColumn, setLogLevelColumn] = useState<SelectedColumn>();
+  const [messageColumn, setMessageColumn] = useState<SelectedColumn>();
+  const [liveView, setLiveView] = useState<boolean>(false);
+  const [orderBy, setOrderBy] = useState<OrderBy[]>([]);
+  const [limit, setLimit] = useState<number>(100);
+  const [filters, setFilters] = useState<Filter[]>([]);
+  const labels = allLabels.components.LogsQueryBuilder;
+
+  useEffect(() => {
+    if (!builderOptions) {
+      return;
+    }
+
+    builderOptions.meta?.otelEnabled !== undefined && setOtelEnabled(builderOptions.meta.otelEnabled);
+    builderOptions.meta?.otelVersion && setOtelVersion(builderOptions.meta.otelVersion);
+    setTimeColumn(getColumnByHint(builderOptions, ColumnHint.Time));
+    setLogLevelColumn(getColumnByHint(builderOptions, ColumnHint.LogLevel));
+    setMessageColumn(getColumnByHint(builderOptions, ColumnHint.LogMessage));
+    builderOptions.columns && setSelectedColumns(builderOptions.columns.filter(c => c.hint === undefined));
+    builderOptions.meta?.liveView && setLiveView(builderOptions.meta.liveView);
+    builderOptions.orderBy && setOrderBy(builderOptions.orderBy);
+    builderOptions.limit && setLimit(builderOptions.limit);
+    builderOptions.filters && setFilters(builderOptions.filters);
+
+    // Run on load
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    const nextColumns = selectedColumns.slice();
+    if (timeColumn) {
+      nextColumns.push(timeColumn);
+    }
+    if (logLevelColumn) {
+      nextColumns.push(logLevelColumn);
+    }
+    if (messageColumn) {
+      nextColumns.push(messageColumn);
+    }
+
+    onBuilderOptionsChange({
+      ...builderOptions,
+      mode: BuilderMode.List,
```

Comment:
maybe we should just set `List` as default value for the `mode`

Reply:
I'm not sure if `mode` is needed anymore, I think it can be determined based on the `builderOptions` and column hints. I think any code that checks `mode` should be replaced with a helper function such as `isListQuery` or `isAggregateQuery`. I've kept it for now so that I don't break too much code in the sql generator